### PR TITLE
Use amp-selector in seatmap-sample

### LINKED ISSUE #1602: Update seatmap to use amp-selector

### PR DESCRIPTION
**Pull request by kul3r4**
This removes the need to bind to class for styling selected and unselected places.
Fixes #1602 
I can work on a second version where I show how to implement a seatmap with svg images for accessible seats where it's not possible to style selected and unselected from css.
@choumx FYI

### CODE REVIEW

File: src/30_Advanced/SeatMap.html [hunk 1]
```diff
@@ -108,29 +116,26 @@ <h1>Seat map</h1>
...
-              }) {{/available}}"></rect>  
+          <amp-selector multiple
+                on="select:AMP.setState({
+                    selectedSeats: event.selectedOptions})">
+            <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+            {{#seats}}                
+              <rect class="seat {{unavailable}}" 
+                x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}" 
+                role="button" 
+                tabindex="0" rx="{{rx}}" ry="{{ry}}" option="{{id}}"></rect>
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: Should we remove `option="{{id}}"` and `role="button"  tabindex="0"` for unavailable seats?

Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: Something like:
`
            {{#seats}}                
              <rect class="seat {{unavailable}}" 
                x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}" 
                {{#available}}
                role="button" 
                option="{{id}}"
                {{/available}}
                tabindex="0" rx="{{rx}}" ry="{{ry}}" ></rect>  
            {{/seats}}
`
will not work. I can duplicate the rects to make it working.

File: src/30_Advanced/SeatMap.html [hunk 2]
```diff
@@ -60,17 +62,23 @@
...
+      #pan-zoom > amp-selector [option][selected] {
+          outline: none;
       }
       .list-container {
         position: relative;
         height: 50vh;
         min-width: 100%;
         box-sizing: border-box;
       }
+      .svg-placeholder {
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: You don't need to create a new class for this, you can target via: 

`[placeholder] > svg`

File: src/30_Advanced/SeatMap.html [hunk 3]
```diff
@@ -88,12 +96,12 @@
...
     <!-- 
-    We combine `amp-list`, `amp-bind` and `amp-pan-zoom` to implement a seatmap:
+    We combine `amp-list`, `amp-bind`, `amp-pan-zoom` and `amp-selector` to implement a seatmap:
 
     * `amp-list` is used to keep the availability of the seatmap up to date.
-    * `amp-bind` is used to update the status of the seats: when a seat is selected or unselected we update the color and 
-    we add the seat id to the `seatmap.selectedSeats` variable. 
-    `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
+    * `amp-bind` is used to update the status of the seats: when a seat is selected we add the seat id to the `selectedSeats` variable. 
+    * `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: s/make possible to/enable

File: src/30_Advanced/SeatMap.html [hunk 4]
```diff
@@ -108,29 +116,26 @@ <h1>Seat map</h1>
...
-              x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}" 
-              role="button" 
-              tabindex="0" rx="{{rx}}" ry="{{ry}}"
-              on="{{#available}} tap:AMP.setState({
-                  selectedSeats: !selectedSeats.includes('{{id}}') ? 
-                  selectedSeats.concat('{{id}}') : copyAndSplice(selectedSeats, selectedSeats.indexOf('{{id}}'), 1)
-              }) {{/available}}"></rect>  
+          <amp-selector multiple
+                on="select:AMP.setState({
+                    selectedSeats: event.selectedOptions})">
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: nit: format the action -> add new line before the closing `}`

File: src/30_Advanced/SeatMap.html [hunk 5]
```diff
@@ -108,29 +116,26 @@ <h1>Seat map</h1>
...
+            <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+            {{#seats}}                
+              <rect class="seat {{unavailable}}"
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: Can you try to order the attributes by importance (e.g. most important first). In this case:

`option button tabindex class x y ...`

File: src/30_Advanced/SeatMap.html [hunk 6]
```diff
@@ -41,7 +43,7 @@
         fill: var(--abe-light-pink);
         pointer-events: none;
       }
-      .selected {
+      rect[selected].seat {
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: Can you target this more specifically.

File: src/30_Advanced/SeatMap.html (hunk 2 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: done

File: src/30_Advanced/SeatMap.html (hunk 3 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: done

File: src/30_Advanced/SeatMap.html (hunk 4 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: done

File: src/30_Advanced/SeatMap.html (hunk 5 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: done

File: src/30_Advanced/SeatMap.html (hunk 6 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: done

**Comment by kul3r4** [comment]
Thanks for the feedback, I have changed the CSS accordingly.

File: src/30_Advanced/SeatMap.html [hunk 7]
```diff
@@ -41,36 +43,42 @@
...
-      .seat-map-svg {
-        box-sizing: border-box;
-        padding-top: 30px;
-      }
       /* `amp-pan-zoom` has a default display: flex to center positioning the child. 
       In this sample we are going to use a seatmap placeholder and for correctly positioning it, we are overwriting the
       `amp-pan-zoom` display to `initial` */
-      #pan-zoom {
+      .seatmap {
         display: initial;
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: Just noticed this, why do we need this? Shouldn't this be the default?

File: src/30_Advanced/SeatMap.html [hunk 8]
```diff
@@ -41,36 +43,42 @@
         fill: var(--abe-light-pink);
         pointer-events: none;
       }
-      .selected {
-        fill: var(--abe-pink);
-      }
       .disabled {
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: move this into `.seatmap-container`

File: src/30_Advanced/SeatMap.html [hunk 9]
```diff
@@ -41,36 +43,42 @@
...
+      }
+      .seatmap [option][selected] {
+        outline: none;
+      }
+      .seatmap >  amp-selector {
+          left: 0;
+          right: 0;
+          margin: auto;
+      }
+      .seatmap-container {
```
Comment (sebastianbenz) [inline src/30_Advanced/SeatMap.html]: move this before `.seatmap` as to match the order in the markup.

File: src/30_Advanced/SeatMap.html (hunk 7 above)
Comment (kul3r4) [reply src/30_Advanced/SeatMap.html]: It's explained above, display initial is not the default css for amp-pan-zoom.

Comment (sebastianbenz) [reply src/30_Advanced/SeatMap.html]: Sorry, should've read that

**Pull request merged**